### PR TITLE
Add localized languages support

### PR DESCRIPTION
**Pull request by medied**
https://phabricator.wikimedia.org/T294392

## Objective

The goal is to allow users to search for langauges using localized strings. That is, if the current user language is english, then typing "hebrew" should render language `he`; if user langauge is spanish, then "hebreo" should render `he`

Is `!! Intl.DisplayNames` an appropriate way to check if the browser supports https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames?

I have tested this in Chrome, Firefox and Safari

### CODE REVIEW

**Review by stephanebisson — APPROVED** [review]
I'll let Huei have a look before merging.

File: src/link/language-selector.js
Comment (hueitan) [inline src/link/language-selector.js:50]: in the example given in mdn, still possible the language has `-`

```js
// Get display names of language in English
let languageNames = new Intl.DisplayNames(['en'], {type: 'language'});
languageNames.of('fr');      // "French"
languageNames.of('de');      // "German"
languageNames.of('fr-CA');   // "Canadian French"
languageNames.of('zh-Hant'); // "Traditional Chinese"
languageNames.of('en-US');   // "American English"
languageNames.of('zh-TW');   // "Chinese (Taiwan)"]
```

Comment (medied) [reply src/link/language-selector.js]: You're right, it turns out there are a certain number of languages which make the `localizedName.of( language )` throw with an error. The invalid languages and error are:

```
['bat-smg', 'be-x-old', 'cbk-zam', 'fiu-vro', 'map-bms', 'roa-rup', 'zh-classical', 'zh-min-nan', 'zh-yue', 'zh-cdo'];
```

```
Uncaught RangeError: invalid value "bat-smg" for option language
    getLocalized language-selector.js:35
    filtered language-selector.js:52
    filterLanguages language-selector.js:48
    onChange components.min.js:10
```

It would be really nice to have some type of built-in support with `Intl` that would let us know if a language is invalid before hand. I looked into [supportedLocalesOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/supportedLocalesOf) and [resolvedOptions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/resolvedOptions) for example, but no luck. One solution is selectively skipping these 10 edge cases, which my last commit shows, though I'm curious to know if there's a better way.

Comment (hueitan) [reply src/link/language-selector.js]: is that possible there will be more and how do you get these 10 cases? 
If you can confirm then I don't see any other solution.

Comment (medied) [reply src/link/language-selector.js]: I narrowed it down to these 10 cases manually by testing, but I don't know what's the probability of new "invalid" languages being added in the future. Looking at the [source commit history](https://github.com/mdn/content/commits/main/files/en-us/web/javascript/reference/global_objects/intl/displaynames), it's about 1-2 commits per months it seems. I checked the listed [issues](https://github.com/mdn/content/issues) also, nothing stands out.

**Review by hueitan — APPROVED** [review]
Let's look into this again when we find other language problem.

**Review by stephanebisson — CHANGES_REQUESTED** [review]
I'm not sure that component state is the right place for the storing the display name support. It never changes for the life of the component or application. I'm not even sure it's worth storing since running `if ( Intl.DisplayNames ) {...}` is basically the same as `if ( displayNamesSupport ) {...}`.

Also, all the language filtering, conversion, including the list of "invalid", could be moved to './languages' to keep this component a little lighter.

**Comment by medied** [comment]
Updated

File: src/link/languages.js
Comment (stephanebisson) [inline src/link/languages.js:401]: I would suggest keeping the check as close as possible to where Intl.DisplayNames is being used so it is easier to keep track that those 2 needs are needed together.

File: src/link/languages.js
Comment (stephanebisson) [inline src/link/languages.js:389]: Instead of passing `setValue` and `setItems`, could you set the value before calling this function, return the items and set those after calling this function? It's uncommon to have a util set state back into a component.

**Review by stephanebisson — CHANGES_REQUESTED** [review]
This is getting so much cleaner. Just a few more details.

File: src/link/languages.js
Comment (medied) [reply src/link/languages.js:389]: Yea of course makes sense

**Review by stephanebisson — APPROVED** [review]

**Pull request merged**
